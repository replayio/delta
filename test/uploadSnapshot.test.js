import { describe, it, expect } from "vitest";
import path from "path";
import fs from "fs";
import fetch from "node-fetch";
import omit from "lodash/omit";

async function uploadImage(image, projectId, branch) {
  let res;

  try {
    res = await fetch("http://localhost:3000/api/uploadSnapshot", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ image, projectId, branch }),
    });

    // const json = await body.json();
    if (res.status !== 200) {
      const body = await res.text();
      console.log(res.status, body);
      return body;
    }
    const body = await res.json();
    return body;
  } catch (e) {
    console.error("error", e);
  }
}

describe("uploadSnapshot", () => {
  it("should upload a snapshot", async () => {
    const projectId = "dcb5df26-b418-4fe2-9bdf-5a838e604ec4";
    const branch = "visuals9";

    const file = "object-collapsed.png";

    const image = {
      content: fs.readFileSync(path.join(__dirname, `images/${file}`), {
        encoding: "base64",
      }),
      file: `./playwright/visuals/dark/${file}`,
    };

    const { data } = await uploadImage(image, projectId, branch);
    expect(omit(data, ["created_at", "id"])).toMatchSnapshot();
  });
});
