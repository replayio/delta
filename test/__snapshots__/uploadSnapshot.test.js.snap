// Vitest Snapshot v1

exports[`uploadSnapshot > should upload a snapshot 1`] = `
{
  "action_changed": null,
  "action_id": "684b2d8d-2ed0-40c7-a2d0-2f351b38c60b",
  "changed": null,
  "file": "./playwright/visuals/dark/object-collapsed.png",
  "path": "dcb5df26-b418-4fe2-9bdf-5a838e604ec4/68e4dd6b357ac328ce5ee7d4f032d6828cc1de32a32a303d8e2aecbd04fede90.png",
  "primary_changed": true,
  "primary_diff_path": "dcb5df26-b418-4fe2-9bdf-5a838e604ec4/ec18c80d8e7d03673f333ead45b30bc06a769fa30168a8289f60fc55ceeb1c7b.png",
  "primary_num_pixels": null,
  "sha": "68e4dd6b357ac328ce5ee7d4f032d6828cc1de32a32a303d8e2aecbd04fede90",
  "status": "Duplicate",
}
`;
